lib/Clam/CfgBuilder.cc

```diff
@@ -320,10 +322,90 @@ static Optional<lin_cst_t> cmpInstToCrabInt(CmpInst &I, crabLitFactory &lfac,
     if (I.getPredicate() == CmpInst::ICMP_ULT) {
       cst.set_unsigned();
     }
-    return cst;
+    return {cst, llvm::None};
     break;
   }
   case CmpInst::ICMP_ULE:
+  case CmpInst::ICMP_ULT: {
+    // translating unsigned comparisons by using select operators
+    // x cmp y, where cmp will be either <= or <
+    // will be translated into:
+    // if x >= 0
+    //     if y >= 0 then x cmp y
+    //     else // x >= 0 y < 0
+    //        false
+    // else
+    //     if y >= 0 then // x < 0 y >= 0
+    //        true
+    //     else x cmp y
+    var_t res = lfac.mkBoolVar();
+    lin_cst_t cmp;
+    if (I.getPredicate() == CmpInst::ICMP_ULE)
+      cmp = lin_cst_t(op0 <= op1);
+    else
+      cmp = lin_cst_t(op0 <= op1 - number_t(1));
+    lin_cst_t cst_op0 = lin_cst_t(op0 >= 0);
+    lin_cst_t cst_op1 = lin_cst_t(op1 >= 0);
+    if (cst_op0.is_tautology()) {
```

Comment:
Add a comment that this is a special case where we know for sure that the first operand is >= 0